mitodl/open-discussions
# change video duration model field, make the migration for it

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #3478

#### What's this PR do?

fixes a schema issue that was causing errors when importing youtube videos. In particular, the duration field we had set as `VARCHAR(10)`, but the [ISO 8601 duration string](https://en.wikipedia.org/wiki/ISO_8601#Durations) returned by the YouTube API can be up to 11 characters long.

These strings look like this:

```ts
`PT${numHours}H${numMinutes}M${numSeconds}S`
```

There are some videos on our list to import which are more than 10 hours long, so we'll have a duration string like `"PT17H28M57S"`, which is 11 characters long. I don't think we'll need more than 11 characters because 1) after setting it to 11 I am able to import all the videos we currently import and 2) youtube currently [doesn't allow uploading videos longer than 12 hours](https://support.google.com/youtube/answer/71673?hl=en&co=GENIE.Platform%3DDesktop) (I believe there are longer videos on the platform which were previously uploaded though).

#### How should this be manually tested?

Make sure you can reproduce the issue:

- On `master`, run the `backpopulate_youtube_videos` management command. It should exit at some point with an error similar to the one in the linked Sentry issue.

then check out this branch, run the migration, and confirm that it fixes the issue and the import can exit without erroring.

When I ran it locally, on `master` I was able to get through 2227 videos before it errored, and then after fixing the issue it pulled down 9001 before doing so. I don't know if the order is deterministic, so your results may be different.
